libs/server/Storage/Session/ObjectStore/SetOps.cs

```diff
@@ -429,5 +432,150 @@ public GarnetStatus SetMembers<TObjectContext>(byte[] key, ArgSlice input, ref G
+                _ = txnManager.Run(true);
+            }
+
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            try
+            {
+                (members, var status) = _setDiff(keys, ref objectStoreLockableContext);
+                if (status == GarnetStatus.WRONGTYPE)
+                {
+                    return status;
+                }
+            }
+            finally
+            {
+                if (createTransaction)
+                    txnManager.Commit(true);
+            }
+
+            return GarnetStatus.OK;
+        }
+
+        /// <summary>
+        /// Diff result store.
+        /// Returns the number of result set.
+        /// </summary>
+        /// <param name="key">destination</param>
+        /// <param name="keys"></param>
+        /// <param name="count"></param>
+        /// <returns></returns>
+        public GarnetStatus SetDiffStore(byte[] key, ArgSlice[] keys, out int count)
+        {
+            count = default;
+
+            if (key.Length == 0 || keys.Length == 0)
+                return GarnetStatus.OK;
+
+            var destination = scratchBufferManager.CreateArgSlice(key);
+
+            var createTransaction = false;
+
+            if (txnManager.state != TxnState.Running)
+            {
+                Debug.Assert(txnManager.state == TxnState.None);
+                createTransaction = true;
+                txnManager.SaveKeyEntryToLock(destination, true, LockType.Exclusive);
+                foreach (var item in keys)
+                    txnManager.SaveKeyEntryToLock(item, true, LockType.Shared);
+                _ = txnManager.Run(true);
+            }
+
+            var objectStoreLockableContext = txnManager.ObjectStoreLockableContext;
+
+            try
+            {
+                (var diffSet, var status) = _setDiff(keys, ref objectStoreLockableContext);
+
+                if (status == GarnetStatus.WRONGTYPE)
+                {
+                    return status;
+                }
+
+                var asMembers = new ArgSlice[diffSet.Count];
+                for (var i = 0; i < diffSet.Count; i++)
+                {
+                    asMembers[i] = scratchBufferManager.CreateArgSlice(diffSet.ElementAt(i));
+                }
+
+                var newSetObject = new SetObject();
+                foreach (var item in diffSet)
+                {
+                    newSetObject.Set.Add(item);
+                    newSetObject.UpdateSize(item);
+                }
+                _ = SET(key, newSetObject, ref objectStoreLockableContext);
+                count = diffSet.Count;
+            }
+            finally
+            {
+                if (createTransaction)
+                    txnManager.Commit(true);
+            }
+
+            return GarnetStatus.OK;
+        }
+
+        private (HashSet<byte[]>, GarnetStatus) _setDiff<TObjectContext>(ArgSlice[] keys, ref TObjectContext objectContext)
```

Comment:
This is not how we interact with Garnet objects... You need to call the StorageSession API and implement the object-specific logic in SetObjectImpl.cs. Please review other methods in this class for reference.